src/components/book-content-mixin.cjsx

```diff
@@ -29,6 +33,36 @@ module.exports =
       else
         img.onload = sizeImage
 
+  getMediaTag: (media) ->
+    # form media tag text based on tag name or data-type
+    tag = media.tagName
+    tag = media.dataset.type if media.dataset.type?
+
+    # capitalize
+    # TODO find one place for all such string formating helpers to go
+    tag = tag.charAt(0).toUpperCase() + tag.substring(1).toLowerCase()
+
+  isMediaLink: (link) ->
+    link.innerText is '[link]' and link.hash.length > 0 and link.hash.search('/') is -1
```

Comment:
Right. I think the 4 cases are:

1. `#id123` and `id123` is on the page (link within current step)
1. `#id123` and `id123` is not on the page (link to ref book but use the `page_id` in new tab)
1. `[UUID][@VER]#id123` links to ref book in new tab
1. `https?://...` external link in new tab